src/main/java/com/google/cloud/spanner/jdbc/JdbcConnection.java

```diff
@@ -402,6 +404,69 @@ public String getSchema() throws SQLException {
     return "";
   }
 
+  @Override
+  public SavepointSupport getSavepointSupport() {
+    return getSpannerConnection().getSavepointSupport();
+  }
+
+  @Override
+  public void setSavepointSupport(SavepointSupport savepointSupport) throws SQLException {
+    checkClosed();
+    try {
+      getSpannerConnection().setSavepointSupport(savepointSupport);
+    } catch (SpannerException e) {
+      throw JdbcSqlExceptionFactory.of(e);
+    }
+  }
+
+  @Override
+  public Savepoint setSavepoint() throws SQLException {
+    checkClosed();
+    try {
+      JdbcSavepoint savepoint = JdbcSavepoint.unnamed();
```

Comment:
What's your take on `inversion of control` ? Are you intentionally using `JdbcSavepoint` since you require the static methods and the `internalGetSavepointName()` method? 

Usually with interfaces we don't expose the implementation class at time of dependency injection. Over here it's not really dependency injection, but would it not be better to use just `Savepoint` interface instead of referencing `JdbcSavepoint` class?